src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,30 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+
+interface Props {
+  todo: Todo;
+  onClose: () => void;
+}
+
+export const TodoModal: React.FC<Props> = ({ todo, onClose }) => {
+  const [todoLoading, setTodoLoading] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    setTodoLoading(true);
+    getUser(todo.userId)
+      .then(setUser)
+      .finally(() => setTodoLoading(false));
+  }, []);
```

Comment:
The `useEffect` dependency array is empty, which means the effect will only run once when the component mounts. If the `todo` prop changes, the user data will not be re-fetched. Consider adding `todo.userId` to the dependency array to ensure the effect runs whenever the `todo` changes.